test/functional/diff/screen_spec.lua

```diff
@@ -0,0 +1,785 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+
+local feed = helpers.feed
+local clear = helpers.clear
+local write_file = helpers.write_file
+
+describe('Diff mode screen', function()
+  local fname = 'Xtest-functional-diff-screen-1'
+  local fname_2 = fname .. '.2'
+  local screen
+
+  local reread = function()
+    feed(':e<cr><c-w>w:e<cr><c-w>w')
+  end
+
+  setup(function()
+    clear()
+    os.remove(fname)
+    os.remove(fname_2)
+  end)
+
+  teardown(function()
+    os.remove(fname)
+    os.remove(fname_2)
+  end)
+
+  before_each(function()
+    clear()
+    feed(':e ' .. fname_2 .. '<cr>')
+    feed(':vnew ' .. fname .. '<cr>')
+    feed(':diffthis<cr>')
+    feed('<c-w>w:diffthis<cr><c-w>w')
+
+    screen = Screen.new(40, 16)
+    screen:attach()
+    screen:set_default_attr_ids({
+      [1] = {foreground = Screen.colors.DarkBlue, background = Screen.colors.WebGray},
+      [2] = {background = Screen.colors.LightCyan1, bold = true, foreground = Screen.colors.Blue1},
+      [3] = {reverse = true},
+      [4] = {background = Screen.colors.LightBlue},
+      [5] = {foreground = Screen.colors.DarkBlue, background = Screen.colors.LightGrey},
+      [6] = {bold = true, foreground = Screen.colors.Blue1},
+      [7] = {bold = true, reverse = true},
+      [8] = {bold = true, background = Screen.colors.Red},
+      [9] = {background = Screen.colors.LightMagenta},
+    })
+  end)
+
+  it('Add a line in beginning of file 2', function()
+    if helpers.pending_win32(pending) then return end
```

Comment:
Won't this work on Windows/appveyor?

Reply:
Ups, I even didn't know what it meant. Let me try.